lib/pluto.js

```diff
@@ -1,148 +1,141 @@
-const _ = require('underscore')
-const MAX_CONSTRUCTOR_ARGUMENTS = 8
+'use strict'
+
+const co = require('co')
+const memoize = require('lodash.memoize')
+
+function isPromise(obj) {
+  return obj && obj.then && typeof obj.then === 'function'
+}
+
+function pluto() {
+  const namesToResolvers = new Map()
 
-function createModule(createModuleCallback) {
   function createInstanceResolver(instance) {
     return function () {
-      return instance
+      if (isPromise(instance)) {
+        return instance
+      } else {
+        return Promise.resolve(instance)
+      }
```

Comment:
I think you can just return `Promise.resolve(instance)` without having to check if it's a promise or not.